### ISSUE TITLE
[Bug]: Since 2.3.0 container crashes when connecting or playing local media with android app

### ISSUE THREAD

**Issue by santschi** [advplyr/audiobookshelf#1912]
### Describe the issue

Since v2.3.0 The container crashes every time the android app tries to save play state.
The web player/interface work as always.

Further I can no longer look at a series in the app (unsure if this is the same issue since the server doesn't crash)

### Steps to reproduce the issue

1. Open Android app (most likely the server crashes here for the first time)
2. Wait for container restart
3. Start playing any downloaded audio book.
4. Either wait until automatic progress save or pause to trigger immediate save
5. The server crashes without error output.

Last log entry:
`
[PlaybackSessionManager] Inserting new session for "<omitted>" (play_local_31BunchMoreNumbers)
`

### Audiobookshelf version

v2.3.0

### Android App Version
v0.9.65-beta

### How are you running audiobookshelf?

Docker

**Comment by advplyr** [comment]
I just fixed the series issue for a patch I will put out today. That is reported in #1914

For the downloaded books, unfortunately you will have to re-download them for now because it will take longer to release a patch on the mobile apps.

**Comment by advplyr** [comment]
@santschi If you can get the log from the actual container for that crash that would be helpful. The full crash stack trace will be in the logs for the container

**Comment by santschi** [comment]
> @santschi If you can get the log from the actual container for that crash that would be helpful. The full crash stack trace will be in the logs for the container

I took a look at the log output of the container itself, which outputs nothing after announcing it is healthy.

Further I took a look at the daily log file, but there was literally nothing else in there. (Start-up, debug message I posted and start up again)
No error or warning, certainly no stack trace.

If you are talking about a different log, or if there is some debug flag (besides the drop-down on the web ui for logs) I can enable I will gladly provide a log.

**Comment by mantas3** [comment]
I have same issue, container crashes when opening android app. Managed to pull some logs from docker:

```
[2023-07-16 23:39:36] INFO: === Starting Server ===
[2023-07-16 23:39:36] INFO: [Server] Init v2.3.0
[2023-07-16 23:39:36] INFO: [Database] Initializing db at "/config/absdatabase.sqlite"
[2023-07-16 23:39:37] INFO: [Database] Db connection was successful
[2023-07-16 23:39:37] INFO: [Database] Db initialized with models: user, library, libraryFolder, book, podcast, podcastEpisode, libraryItem, mediaProgress, series, bookSeries, author, bookAuthor, collection, collectionBook, playlist, playlistMediaItem, device, playbackSession, feed, feedEpisode, setting
[2023-07-16 23:39:38] INFO: [Database] Db data loaded in 551ms
[2023-07-16 23:39:38] INFO: [BackupManager] 0 Backups Found
[2023-07-16 23:39:38] INFO: [BackupManager] Auto Backups are disabled
[2023-07-16 23:39:38] INFO: [LogManager] Init current daily log filename: 2023-07-16.txt
[2023-07-16 23:39:38] INFO: [Watcher] Initializing watcher for "knygos".
[2023-07-16 23:39:38] INFO: Listening on port :83
[2023-07-16 23:39:38] INFO: [Watcher] "knygos" Ready
[2023-07-16 23:39:39] INFO: [Server] Socket Connected Johsl7YAU5IozZY0AAAB
[2023-07-16 23:39:42] INFO: [Server] Socket Connected wK3Moax4goCBXiCGAAAD
[2023-07-16 23:39:48] INFO: [Server] Socket wK3Moax4goCBXiCGAAAD disconnected from client "mantas" after 5363ms (Reason: transport close)
[2023-07-16 23:39:54] INFO: [Server] Socket Connected erwKulQZrkj6Kc5CAAAF
[2023-07-16 23:39:54] INFO: [PlaybackSessionManager] Syncing local session "Orwell Collection" (play_local_ad63e189-4319-4617-b16e-b0ac299bbca7)
node:internal/process/promises:279
            triggerUncaughtException(err, true /* fromPromise */);
            ^

Error
    at Database.<anonymous> (/node_modules/sequelize/lib/dialects/sqlite/query.js:185:27)
    at /node_modules/sequelize/lib/dialects/sqlite/query.js:183:50
    at new Promise (<anonymous>)
    at Query.run (/node_modules/sequelize/lib/dialects/sqlite/query.js:183:12)
    at /node_modules/sequelize/lib/sequelize.js:315:28
    at async SQLiteQueryInterface.insert (/node_modules/sequelize/lib/dialects/abstract/query-interface.js:308:21)
    at async PlaybackSession.save (/node_modules/sequelize/lib/model.js:2490:35)
    at async Function.create (/node_modules/sequelize/lib/model.js:1362:12)
    at async PlaybackSessionManager.syncLocalSession (/server/managers/PlaybackSessionManager.js:109:7)
    at async PlaybackSessionManager.syncLocalSessionsRequest (/server/managers/PlaybackSessionManager.js:83:22) {
  name: 'SequelizeForeignKeyConstraintError',
  parent: [Error: SQLITE_CONSTRAINT: FOREIGN KEY constraint failed] {
    errno: 19,
    code: 'SQLITE_CONSTRAINT',
    sql: 'INSERT INTO `playbackSessions` (`id`,`mediaItemType`,`displayTitle`,`displayAuthor`,`duration`,`playMethod`,`mediaPlayer`,`startTime`,`currentTime`,`serverVersion`,`coverPath`,`timeListening`,`mediaMetadata`,`extraData`,`createdAt`,`updatedAt`,`userId`,`deviceId`,`libraryId`) VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19);'
  },
  original: [Error: SQLITE_CONSTRAINT: FOREIGN KEY constraint failed] {
    errno: 19,
    code: 'SQLITE_CONSTRAINT',
    sql: 'INSERT INTO `playbackSessions` (`id`,`mediaItemType`,`displayTitle`,`displayAuthor`,`duration`,`playMethod`,`mediaPlayer`,`startTime`,`currentTime`,`serverVersion`,`coverPath`,`timeListening`,`mediaMetadata`,`extraData`,`createdAt`,`updatedAt`,`userId`,`deviceId`,`libraryId`) VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19);'
  },
  sql: 'INSERT INTO `playbackSessions` (`id`,`mediaItemType`,`displayTitle`,`displayAuthor`,`duration`,`playMethod`,`mediaPlayer`,`startTime`,`currentTime`,`serverVersion`,`coverPath`,`timeListening`,`mediaMetadata`,`extraData`,`createdAt`,`updatedAt`,`userId`,`deviceId`,`libraryId`) VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19);',
  parameters: {},
  table: undefined,
  fields: undefined,
  value: undefined,
  index: undefined,
  reltype: undefined
}
Config /config /metadata
```

**Comment by advplyr** [comment]
@mantas3 I'm fixing this in the upcoming patch to the server

**Comment by mantas3** [comment]
Until that reinstalling mobile app seems to help.

**Comment by advplyr** [comment]
The issue is with playing downloaded books. There still may be some issues until the patch

**Comment by advplyr** [comment]
Fixed in [v2.3.1](https://github.com/advplyr/audiobookshelf/releases/tag/v2.3.1)

**Comment by PyroghostX** [comment]
@advplyr I am on 2.3.1 and the container is crashing






[2023-07-17 15:01:53] INFO: [Watcher] Initializing watcher for "Audiobooks".
[2023-07-17 15:01:53] INFO: Listening on port :80
[2023-07-17 15:01:55] INFO: [Watcher] "Audiobooks" Ready
node:internal/process/promises:279
            triggerUncaughtException(err, true /* fromPromise */);
            ^
Error
    at Database.<anonymous> (/node_modules/sequelize/lib/dialects/sqlite/query.js:185:27)
    at /node_modules/sequelize/lib/dialects/sqlite/query.js:183:50
    at new Promise (<anonymous>)
    at Query.run (/node_modules/sequelize/lib/dialects/sqlite/query.js:183:12)
    at /node_modules/sequelize/lib/sequelize.js:315:28
    at async SQLiteQueryInterface.insert (/node_modules/sequelize/lib/dialects/abstract/query-interface.js:308:21)
    at async PlaybackSession.save (/node_modules/sequelize/lib/model.js:2490:35)
    at async Function.create (/node_modules/sequelize/lib/model.js:1362:12)
    at async PlaybackSessionManager.syncLocalSession (/server/managers/PlaybackSessionManager.js:139:7)
    at async PlaybackSessionManager.syncLocalSessionRequest (/server/managers/PlaybackSessionManager.js:189:20) {
  name: 'SequelizeForeignKeyConstraintError',
  parent: [Error: SQLITE_CONSTRAINT: FOREIGN KEY constraint failed] {
    errno: 19,
    code: 'SQLITE_CONSTRAINT',
    sql: 'INSERT INTO `playbackSessions` (`id`,`mediaItemId`,`mediaItemType`,`displayTitle`,`displayAuthor`,`duration`,`playMethod`,`mediaPlayer`,`startTime`,`currentTime`,`serverVersion`,`coverPath`,`timeListening`,`mediaMetadata`,`extraData`,`createdAt`,`updatedAt`,`userId`,`deviceId`,`libraryId`) VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19,$20);'
  },
  original: [Error: SQLITE_CONSTRAINT: FOREIGN KEY constraint failed] {
    errno: 19,
    code: 'SQLITE_CONSTRAINT',
    sql: 'INSERT INTO `playbackSessions` (`id`,`mediaItemId`,`mediaItemType`,`displayTitle`,`displayAuthor`,`duration`,`playMethod`,`mediaPlayer`,`startTime`,`currentTime`,`serverVersion`,`coverPath`,`timeListening`,`mediaMetadata`,`extraData`,`createdAt`,`updatedAt`,`userId`,`deviceId`,`libraryId`) VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19,$20);'
  },
  sql: 'INSERT INTO `playbackSessions` (`id`,`mediaItemId`,`mediaItemType`,`displayTitle`,`displayAuthor`,`duration`,`playMethod`,`mediaPlayer`,`startTime`,`currentTime`,`serverVersion`,`coverPath`,`timeListening`,`mediaMetadata`,`extraData`,`createdAt`,`updatedAt`,`userId`,`deviceId`,`libraryId`) VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19,$20);',
  parameters: {},
  table: undefined,
  fields: undefined,
  value: undefined,
  index: undefined,
  reltype: undefined
}

**Comment by santschi** [comment]
I got a stack trace as well now. No idea what changed but this is mine:
```
[2023-07-17 18:18:18] INFO: [Watcher] "Audiobooks" Ready
[2023-07-17 18:18:48] INFO: [Server] Socket Connected -wPMRjHRj0wAK_XwAAAB
[2023-07-17 18:18:48] DEBUG: [Server] User Online admin (SocketAuthority.js:162)
[2023-07-17 18:19:10] INFO: Received ping
[2023-07-17 18:19:14] DEBUG: [PlaybackSessionManager] Inserting new session for "<omitted>" (04767148-8960-4e96-b2c5-af445308ad32) (PlaybackSessionManager.js:138)
node:internal/process/promises:279
            triggerUncaughtException(err, true /* fromPromise */);
            ^

Error
    at Database.<anonymous> (/node_modules/sequelize/lib/dialects/sqlite/query.js:185:27)
    at /node_modules/sequelize/lib/dialects/sqlite/query.js:183:50
    at new Promise (<anonymous>)
    at Query.run (/node_modules/sequelize/lib/dialects/sqlite/query.js:183:12)
    at /node_modules/sequelize/lib/sequelize.js:315:28
    at async SQLiteQueryInterface.insert (/node_modules/sequelize/lib/dialects/abstract/query-interface.js:308:21)
    at async PlaybackSession.save (/node_modules/sequelize/lib/model.js:2490:35)
    at async Function.create (/node_modules/sequelize/lib/model.js:1362:12)
    at async PlaybackSessionManager.syncLocalSession (/server/managers/PlaybackSessionManager.js:139:7)
    at async PlaybackSessionManager.syncLocalSessionRequest (/server/managers/PlaybackSessionManager.js:189:20) {
  name: 'SequelizeForeignKeyConstraintError',
  parent: [Error: SQLITE_CONSTRAINT: FOREIGN KEY constraint failed] {
    errno: 19,
    code: 'SQLITE_CONSTRAINT',
    sql: 'INSERT INTO `playbackSessions` (`id`,`mediaItemId`,`mediaItemType`,`displayTitle`,`displayAuthor`,`duration`,`playMethod`,`mediaPlayer`,`startTime`,`currentTime`,`serverVersion`,`coverPath`,`timeListening`,`mediaMetadata`,`extraData`,`createdAt`,`updatedAt`,`userId`,`deviceId`,`libraryId`) VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19,$20);'
  },
  original: [Error: SQLITE_CONSTRAINT: FOREIGN KEY constraint failed] {
    errno: 19,
    code: 'SQLITE_CONSTRAINT',
    sql: 'INSERT INTO `playbackSessions` (`id`,`mediaItemId`,`mediaItemType`,`displayTitle`,`displayAuthor`,`duration`,`playMethod`,`mediaPlayer`,`startTime`,`currentTime`,`serverVersion`,`coverPath`,`timeListening`,`mediaMetadata`,`extraData`,`createdAt`,`updatedAt`,`userId`,`deviceId`,`libraryId`) VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19,$20);'
  },
  sql: 'INSERT INTO `playbackSessions` (`id`,`mediaItemId`,`mediaItemType`,`displayTitle`,`displayAuthor`,`duration`,`playMethod`,`mediaPlayer`,`startTime`,`currentTime`,`serverVersion`,`coverPath`,`timeListening`,`mediaMetadata`,`extraData`,`createdAt`,`updatedAt`,`userId`,`deviceId`,`libraryId`) VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19,$20);',
  parameters: {},
  table: undefined,
  fields: undefined,
  value: undefined,
  index: undefined,
  reltype: undefined
}

```

**Comment by santschi** [comment]
> Until that reinstalling mobile app seems to help.

I deleted app cache & storage, deleted all the downloaded audiobooks.
Now everything works perfectly!

@advplyr Thanks for working on this app! It's fantastic.

**Comment by PyroghostX** [comment]
My container is crashing seconds after starting. So not related to any
clients

**Comment by advplyr** [comment]
I believe I figured out why the crash was still happening and I will patch it today. Removing and re-adding the connection will also work

**Comment by advplyr** [comment]
Fixed in [v2.3.2](https://github.com/advplyr/audiobookshelf/releases/tag/v2.3.2)

Let me know if there are still any crashes with local media playing from app
